std/encoding.d

```diff
@@ -2250,24 +2138,27 @@ body
             immutable minReservePlace = 6;
         }
 
-        Dst[] buffer = new Dst[s.length];
```

Comment:
Even with reserve, the old code does less work. Return values of strongly pure functions are implicitly convertible to immutable - could this be leveraged here, using a (possibly static) nested function?